src/aliased_buffer.h

```diff
@@ -152,9 +149,10 @@ class AliasedBuffer {
       return this->operator+=(static_cast<NativeT>(val));
     }
 
-    template <typename T>
-    inline Reference& operator-=(const T& val) {
-      return this->operator+=(-val);
+    inline Reference& operator-=(const NativeT& val) {
+      const NativeT current = aliased_buffer_->GetValue(index_);
+      aliased_buffer_->SetValue(index_, current - val);
+      return *this;
```

Comment:
Just curious, was this causing real issues? Unary minus on unsigned values should not really be an issue afaik, although I agree that being explicit is probably a good idea?

Reply:
No functional issue, I was just seeing a warning in VC++ and looking at the code it's not clear what the behavior is when values are unsigned.  Looking around it appears to be [speced behavior](https://stackoverflow.com/questions/8026694/c-unary-minus-operator-behavior-with-unsigned-operands), but still didn't feel intuitive.